### ISSUE TITLE
Suggestion: use Mutation Observer to hide tooltips on disabled elements

### ISSUE THREAD

**Issue by ghost** [ng-bootstrap/ng-bootstrap#2503]
### Bug description:
Tooltips don't hide when the button becomes disabled after clicking on this button.

This issue was discussed here https://github.com/ng-bootstrap/ng-bootstrap/issues/1250

> Mouse events don't fire on disabled elements (ex.: http://stackoverflow.com/questions/3100319/event-on-a-disabled-input) so we can't do anything on the library side to detect such situation. But you can always close a tooltip manually if you've got a very specific use-case as in your plunk: http://plnkr.co/edit/75P69nN39WIXfCHUEBRJ?p=preview

But I've found a way to fix the issue. It is possible to detect attribute change with MutationObserver.
You can implement MutationObserver in the ngbTooltip and hide tooltips when attribute `disabled` changes to `true`.  MutationObserver is widely supported. And, to be sure, you can always check if browser supports it.

### Link to minimally-working plunker that reproduces the issue:
Bug: https://stackblitz.com/edit/angular-x3smmz?file=src%2Fapp%2Fapp.component.ts
Fix: https://stackblitz.com/edit/angular-1nat52?file=src/app/disable-watcher.directive.ts

**Comment by pkozlowski-opensource** [comment]
My first knee-jerk reaction is that it could work but might be too big of a machinery for such a rare use-case. The thing is that a tooltip can be placed on any element  so I'm not sure if we want to special-case it just for disabled buttons.

I feel like the problem would be even more pronounced when we implement delay (#1052).

tl;dr; I hear what you are saying, I see how it could work but I'm not sure it is worth it...

**Comment by maxkorz** [comment]
I don't think this is a rare case. A lot of people suffer from this issue.
- Angular 1 bootstrap: [1](https://github.com/angular-ui/bootstrap/issues/6385) [2](https://github.com/angular-ui/bootstrap/issues/3354) [3](https://github.com/angular-ui/bootstrap/issues/3167)
- This repo: [1](https://github.com/ng-bootstrap/ng-bootstrap/issues/1975) [2](https://github.com/ng-bootstrap/ng-bootstrap/issues/1250) [3](https://github.com/ng-bootstrap/ng-bootstrap/issues/2471)
- Bootstrap repo: [1](https://github.com/twbs/bootstrap/issues/22380) [2](https://github.com/twbs/bootstrap/issues/21232) [3](https://github.com/twbs/bootstrap/issues/8214)

\+ people who don't have a github account + people who are lazy and won't create an issue.

Anyway, even if you won't fix the problem, I hope other people will find this issue and see my solution.
